### DISCUSSION TITLE
Is the API restarted on every request?

### DISCUSSION THREAD

**Discussion by s4m-mo** [deta/space-docs#117] [Q&A] [answered]
I make a request to my API, and for every single request, the micro seems to be starting up with the opening Uvicorn logs.

My question is, every time I make a REST API request to my Deta Space Micro, does it run the micro from scratch every single time?

**Comment by SlumberDemon** [comment, marked as answer]
Yes, A space app only stays alive for 20s which is its timeout and after that timeout it gets destroyed, this is on a request based system so this only happens when a request is made to that endpoint. This is because deta is a serverless platform.

More on serverless platforms can be found here
> https://www.cloudflare.com/learning/serverless/what-is-serverless/

**Reply by s4m-mo** [reply]
Thank you for the clarification. I have a couple of questions.

- Will any future paid Deta plans allow for a greater timeout?
- Also, do you have any tips on improving python FastAPI startup times?
Could I run a python py_compile task on space push and then execute the `.pyc` file rather than the interpreter having to re-bytecode-compile the source every time?

**Reply by SlumberDemon** [reply]
> Will any future paid Deta plans allow for a greater timeout?

There are no plans for anything like that yet

> Also, do you have any tips on improving python FastAPI startup times?

Not really

> Could I run a python py_compile task on space push and then execute the .pyc file rather than the interpreter having to re-bytecode-compile the source every time?

You can't skip any steps during the push
